File: src/test/java/br/com/leonardoferreira/jirareport/base/CleanDatabase.java
package br.com.leonardoferreira.jirareport.base;

import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

@Slf4j
@Component
public class CleanDatabase {

    @Autowired
    private DataSource dataSource;

    @SneakyThrows
    public void clean() {
        try (Connection connection = dataSource.getConnection();
             Statement stmt = connection.createStatement()) {
            truncateTables(connection, stmt);
        }
    }

    @SneakyThrows
    private void truncateTables(final Connection connection, final Statement stmt) {
        try (ResultSet rs = connection.getMetaData().getTables(null, null, null, new String[]{"TABLE"})) {
            while (rs.next()) {
                String tableName = rs.getString("TABLE_NAME");
                if (!"flyway_schema_history".equals(tableName)) {
                    stmt.executeUpdate("TRUNCATE TABLE " + tableName + " RESTART IDENTITY CASCADE");
                }
            }
        }
    }

}

File: src/test/kotlin/br/com/jiratorio/base/CleanDatabase.kt
package br.com.jiratorio.base

import org.springframework.stereotype.Component
import java.sql.Connection
import java.sql.Statement
import javax.sql.DataSource

@Component
class CleanDatabase(
    private val dataSource: DataSource
) {

    fun clean() {
        dataSource.connection.use { connection ->
            connection.createStatement().use { stmt ->
                truncateTables(connection, stmt)
            }
        }
    }

    private fun truncateTables(connection: Connection, stmt: Statement) {
        connection.metaData.getTables(null, null, null, arrayOf("TABLE")).use {
            while (it.next()) {
                val tableName = it.getString("TABLE_NAME")
                if ("flyway_schema_history" != tableName) {
                    stmt.executeUpdate("TRUNCATE TABLE $tableName RESTART IDENTITY CASCADE")
                }
            }
        }
    }
}
